File: LeetCodeCPlusPlus/py6.cpp
#include<string>
#include<vector>
using namespace std;
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s;

        vector<string> rows(min(numRows, int(s.size())));
        int curRow = 0;
        int direction = -1;  //第一轮从0开始，会变方向，因此设置成反方向

        for (char c : s) {
            rows[curRow] += c;
            if (curRow == 0 || curRow == numRows - 1) direction = -direction;
            curRow += direction;
        }

        string ret;
        for (string row : rows) ret += row;
        return ret;
    }
};

File: LeetCodePython/py6.py
# Z字变换，模拟
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        result = [[] for i in range(numRows)]
      
        direction = 1
        index = 0
        for ch in s:
            result[index].append(ch)
            index = index + direction
            if index == numRows or index == -1:
                direction = - direction
                index = direction*2+index
     
        result = ["".join(item) for item in result]
        result = "".join(result)
        return result
                
                
        